src/main/kotlin/it/pagopa/wallet/scheduler/services/SchedulerLockService.kt

```diff
@@ -0,0 +1,89 @@
+package it.pagopa.wallet.scheduler.service
+
+import it.pagopa.wallet.scheduler.config.properties.RedisJobLockPolicyConfig
+import it.pagopa.wallet.scheduler.exceptions.LockNotAcquiredException
+import it.pagopa.wallet.scheduler.exceptions.LockNotReleasedException
+import it.pagopa.wallet.scheduler.exceptions.SemNotAcquiredException
+import it.pagopa.wallet.scheduler.exceptions.SemNotReleasedException
+import java.util.concurrent.TimeUnit
+import org.redisson.api.RedissonReactiveClient
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Service
+import reactor.core.publisher.Mono
+import reactor.kotlin.core.publisher.switchIfEmpty
+
+@Service
+class SchedulerLockService(
+    @Autowired private val redissonClient: RedissonReactiveClient,
+    @Autowired private val redisJobLockPolicyConfig: RedisJobLockPolicyConfig
+) {
+    private val logger: Logger = LoggerFactory.getLogger(javaClass)
+
+    fun acquireJobLock(jobName: String): Mono<Void> {
+        logger.info("Trying to acquire lock for job: {}", jobName)
+        return redissonClient
+            .getLock(redisJobLockPolicyConfig.getLockNameByJob(jobName))
+            .tryLock(
+                redisJobLockPolicyConfig.waitTimeSec,
+                redisJobLockPolicyConfig.ttlSec,
+                TimeUnit.SECONDS
+            )
+            .filter { it == true } // only lock acquired
+            .switchIfEmpty { Mono.error(LockNotAcquiredException(jobName)) }
+            .doOnSuccess { logger.info("Lock acquired for job: {}", jobName) }
+            .onErrorResume {
+                logger.error("Lock acquiring error for job: {}", jobName, it)
+                Mono.error(LockNotAcquiredException(jobName, it))
+            }
+            .then()
```

Comment:
```suggestion
            .filter { it == true } // only lock acquired
            .doOnSuccess { logger.info("Lock acquired for job: {}", jobName) }
            .onErrorMap {
                logger.error("Lock acquiring error for job: {}", jobName, it)
                LockNotAcquiredException(jobName, it)
            }
            .switchIfEmpty { Mono.error(LockNotAcquiredException(jobName)) }
            .then()
```

Move switchIfEmpty method at the chain end to avoid map the `LockNotAcquiredException` custom exception raised in the switchIfEmpty in the `onErrorResume` method.
Moreover a onErrorMap here could be used to map any exception in a custom one